src/ConnectedMode/StatefulServerBranchProvider.cs

```diff
@@ -81,6 +87,14 @@ public async Task<string> GetServerBranchNameAsync(CancellationToken token)
             return selectedBranch;
         }
 
+        private Task<string> DoGetServerBranchNameAsync(CancellationToken token)
+        {
+            return threadHandling.RunOnBackgroundThread(() =>
+            {
+                return serverBranchProvider.GetServerBranchNameAsync(token);
+            });
```

Comment:
I could have written this as nested lambda expressions, but I find them harder to read, and harder to set breakpoints on each individual step.